### PR TITLE
Enable Checkstyle's `JavadocStyle` module

### PR DESCRIPTION
**Pull request by rickie**
Suggested commit message:
```
Enable Checkstyle's `JavadocStyle` module (#451)

See: 
- https://checkstyle.org/config_javadoc.html#JavadocStyle
- https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.html
```

### CODE REVIEW

**Review by Stephan202 — APPROVED** [review]
:sparkles:

**Review by Badbond — APPROVED** [review]
Might be nice to also enable `checkEmptyJavadoc` as it is disabled by default.

**Comment by Stephan202** [comment]
Good point @Badbond. Adding `checkEmptyJavadoc` SGTM. @rickie shall we add it to this PR?

**Comment by rickie** [comment]
Yes will do!

**Comment by rickie** [comment]
It found a missing one, added a proposal.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/PrimitiveComparisonTest.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
 // XXX: Can we perhaps work-around this by describing the fixes in reverse order?
 final class PrimitiveComparisonTest {
   /**
+   * Identification test for primitive {@code byte} comparisons.
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/PrimitiveComparisonTest.java:15]: Proposal to stick with the usual grammatical form:
```suggestion
   * Tests the identification of {@code byte} comparisons for which boxing can be avoided.
```

**Review by Stephan202 — APPROVED** [review]
:rocket:

**Review by Badbond — APPROVED** [review]
Thanks for the tweakz! :+1:

**Pull request merged**